semanticdb/symtab/src/main/scala/scala/meta/internal/symtab/GlobalSymbolTable.scala

```diff
@@ -0,0 +1,77 @@
+package scala.meta.internal.symtab
+
+import scala.annotation.tailrec
+import scala.collection.concurrent.TrieMap
+import scala.meta.internal.classpath.ClasspathIndex
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.io._
+import scala.meta.internal.metacp._
+import scala.meta.internal.scalacp.Scalalib
+import scala.meta.internal.semanticdb.Scala._
+import scala.reflect.NameTransformer
+
+/** A lazy symbol table that returns global symbols on-the-fly from disk. */
+final class GlobalSymbolTable(classpathIndex: ClasspathIndex) extends SymbolTable {
+
+  private val symbolCache = TrieMap.empty[String, SymbolInformation]
+  Scalalib.synthetics.foreach(enter)
```

Comment:
This takes care of scalalib synthetics, but doesn't take RootPackage and EmptyPackage into account.